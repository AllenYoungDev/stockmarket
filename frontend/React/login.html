<html>
<head>
<style>
.top-bar { display:grid; grid-template-columns: 50% 50%; margin: 0; padding: 1% 0.5% 1% 0; background: rgb(0,0,0); color: rgb(255,255,255); font-family: Arial, Helvetica, sans-serif; }
.top-bar-left-column { grid-row: 1; grid-column: 1; text-align: left; margin:0; }
.top-bar-right-column { grid-row: 1; grid-column: 2; text-align: right; margin:0; }

@media only screen and (orientation: landscape) {
.login-controls-section {background-color: #737CA1;color: rgb(250,250,250);padding-bottom: 1.5%;}
.login-controls-section-heading {text-align: left; font-size: 300%; font-family: Arial, Helvetica, sans-serif; font-weight: bold; margin: 0;}
.login-form {margin: 1% 0% 0% 0%;}
.login-form-div-grid {display:grid; grid-template-columns: 25% 25% 15% 10% 25%; grid-gap:5px; font-size: 130%;font-family: arial;}
.login-form-label-email-address {grid-row: 1; grid-column: 2;text-align: right;}
.login-form-input-email-address {grid-row: 1; grid-column: 3;}
.login-form-label-password {grid-row: 2; grid-column: 2;text-align: right;}
.login-form-input-password {grid-row: 2; grid-column: 3;}
.login-form-password-show-hide {grid-row: 2; grid-column: 4;margin: 0; padding: 0.35em 0 0 0.25em;text-align: left;font-size: 60%;font-family: Arial, Helvetica, sans-serif; font-weight: 50;}
.login-form-div-login-button {text-align: center;}
.login-form-error-notice {text-align: center; margin: 1% 0 0 0;font-size: 120%;color: rgb(70,0,0);}
.login-form-input-login-button {margin: 1% 0 0 0; padding: 0.5%; font-size: 150%; font-weight: 900; background-color: #2B547E; border-radius: 0px; color: rgb(250,250,250);}
}
@media only screen and (orientation: portrait) {
.login-controls-section {background-color: #737CA1;color: rgb(250,250,250);padding-bottom: 1.5%;}
.login-controls-section-heading {text-align: left; font-size: 300%; font-family: Arial, Helvetica, sans-serif; font-weight: bold; margin: 0 0 3% 0;}
.login-form {margin: 1% 0% 0% 0%;font-size: 200%;}
.login-form-div-grid {display:grid; grid-template-columns: 40% 50% 10%; font-size: 130%;font-family: arial;}
.login-form-label-email-address {grid-row: 1; grid-column: 1;text-align: right; font-size: 130%; margin-right: 3%;}
.login-form-input-email-address {grid-row: 1; grid-column: 2; font-size: 130%;margin-bottom: 2%;}
.login-form-label-password {grid-row: 2; grid-column: 1;text-align: right; font-size: 130%; margin-right: 3%;}
.login-form-input-password {grid-row: 2; grid-column: 2; font-size: 130%;}
.login-form-password-show-hide {grid-row: 2; grid-column: 3;margin: 0; padding: 0.35em 0 0 0.25em;text-align: left;font-size: 60%;font-family: Arial, Helvetica, sans-serif; font-weight: 50;}
.login-form-div-login-button {text-align: center;}
.login-form-error-notice {text-align: center; margin: 1% 0 0 0;font-size: 120%;color: rgb(70,0,0);}
.login-form-input-login-button {margin: 1% 0 0 0; padding: 0.5%; font-size: 150%; font-weight: 900; background-color: #2B547E; border-radius: 0px; color: rgb(250,250,250);}
}

.footer   {margin: 0; padding: 0.25% 0.5% 0.1% 0; background: rgb(0,0,0); color: rgb(255,255,255); font-family: Arial, Helvetica, sans-serif; text-align: center;}
</style>
</head>
<body style="margin: 0; background-color: #FFFFFF;color: rgb(250,250,250);">

<div id="login-page"></div>

<script src="https://unpkg.com/react@18/umd/react.development.js" crossorigin></script>
<script src="https://unpkg.com/react-dom@18/umd/react-dom.development.js" crossorigin></script>

<script type="module" src="login.render.js"></script>

<noscript>Your browser must support JavaScript.</noscript>
</body>
</html>