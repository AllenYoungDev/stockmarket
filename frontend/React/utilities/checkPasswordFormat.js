const reLetterChecker = /[a-zA-Z]/;
const reNumberChecker = /[0-9]/;
const reSpecialCharacterChecker = /[\~\`\!\@\#\$\%\^\&\*\-\_\+\=\|\\\/\?\;\:\"\'\<\>\,\.\{\}\[\]\(\)]/;

export function checkPasswordFormat(password) {
    var numberOfCharactersInPassword = password.trim().length;

    if (numberOfCharactersInPassword < 10) {return false;}

    if (password.match(reLetterChecker) === null) {return false;}

    if (password.match(reNumberChecker) === null) {return false;}

    if (password.match(reSpecialCharacterChecker) === null) {return false;}

    return true;
}