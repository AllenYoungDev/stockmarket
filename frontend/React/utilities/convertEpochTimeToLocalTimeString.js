export function convertEpochTimeToLocalTimeString(epochTime) {
    var date = new Date(epochTime);
    return date.toLocaleDateString("default") + " " + date.toLocaleTimeString("default");
}