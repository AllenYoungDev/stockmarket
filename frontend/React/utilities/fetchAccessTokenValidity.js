'use strict';

import '../settings/ServerUrls.js';

export function fetchAccessTokenValidity(setLoginState, ignore) {

    console.log('fetchAccessTokenValidity() start.')

    fetch(window.backendServerUrl + "/CheckAccessTokenValidity", {method: "GET", credentials: 'include'})
    .then((response) => response.text())
    .then((text) => {
        if (text === 'valid') {
            console.log('fetchAccessTokenValidity():  access token is valid.');
            if (ignore) {return;}
            if (document.cookie.includes('admin=true')) {
                setLoginState('admin-login');
            } else {
                setLoginState('non-admin-login');
            }
        } else {
            console.log('fetchAccessTokenValidity():  access token is invalid.');
            if (ignore) {return;}
            setLoginState('no-login');
        }
    })
    .catch((err) => {
        console.log(`fetchAccessTokenValidity():  fetch() network error.  ${err}`);
        if (ignore) {return;}
        setLoginState('fetch-error');
    });
}