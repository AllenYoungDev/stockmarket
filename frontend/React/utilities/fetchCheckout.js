import '../settings/ServerUrls.js';

export function fetchCheckout(numberOfSharesToBuy, errorNoticeDomNodeRef) {
  let checkoutResponse = 0;
  let checkoutResponseStatus = 0;

  fetch(window.backendServerUrl + "/Checkout", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },      
    body: JSON.stringify({'numberOfSharesToBuy': numberOfSharesToBuy}),
    credentials: 'include',
    })
  .then((response) => {
    checkoutResponse = response;
    response.headers.forEach((value, key) => {
      console.log(`CheckoutButton() handleOnClick() checkoutResponse.headers key==>value:  ${key} ==> ${value}`)
    });
    checkoutResponseStatus=response.status;
    return response.text();})
  .then((text) => {
    if(checkoutResponseStatus === 200) {
      //window.location.href = window.frontendServerUrl;
      //This is not the proper implementation.  The page must navigate to /Checkout URL.
    } else {
      errorNoticeDomNodeRef.current.innerHTML = `Error.  Cannot proceed to checkout.  ${text}.`
    }
  })
  .catch((err) => {
    //'fetch-error'
    errorNoticeDomNodeRef.current.innerHTML = `Unable to proceed to checkout.  Network error.`;
  });   
}