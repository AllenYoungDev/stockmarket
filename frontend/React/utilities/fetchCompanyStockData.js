'use strict';

import './ServerUrl.js';

export function fetchCompanyStockData(setStockDataServerResponse, ignore) {

    fetch(window.backendServerUrl + "/GetCompanyStockData", {method: "GET", credentials: 'include'})
    .then((response) => response.status === 200 ? response.json() : response.text())
    .then((data) => {
      console.log(`CompanyStockStats() typeof data:  ${typeof data}.`)
      if (typeof data === 'object') {
        console.log(`CompanyStockStats() JSON response:  ${data}.`)
        //console.log(`CompanyStockStats() data[0]:  ${data[0]}.`)
        if (!ignore) {
          setStockDataServerResponse({responseType: 'success', stockData: data});
        }
      } else {
        console.error(`CompanyStockStats() text response:  ${data}.`)
        if (!ignore) {
          setStockDataServerResponse({responseType: 'server-error', stockData: {}});
        }      
      }
    })
    .catch((err) => {
      console.error(`CompanyStockStats() fetch() error:  ${err}.`)
      if (!ignore) {
        setStockDataServerResponse({responseType: 'fetch-error', stockData: {}});
      }    
    });
}