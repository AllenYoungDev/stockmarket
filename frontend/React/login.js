'use strict';

//import React, { StrictMode } from "react";
//import { createRoot } from 'react-dom/client';

var _slicedToArray = function () { function sliceIterator(arr, i) { var _arr = []; var _n = true; var _d = false; var _e = undefined; try { for (var _i = arr[Symbol.iterator](), _s; !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"]) _i["return"](); } finally { if (_d) throw _e; } } return _arr; } return function (arr, i) { if (Array.isArray(arr)) { return arr; } else if (Symbol.iterator in Object(arr)) { return sliceIterator(arr, i); } else { throw new TypeError("Invalid attempt to destructure non-iterable instance"); } }; }();

import { validateEmail } from './utilities/validators.js';
import './settings/ServerUrls.js';
import { fetchAccessTokenValidity } from './utilities/fetchAccessTokenValidity.js';
import { fetchLogin } from './utilities/fetchLogin.js';
import { checkPasswordFormat } from './utilities/checkPasswordFormat.js';
import { footer } from './footer.js';

/* *************************************************************************************************************
LoginPage React component
************************************************************************************************************** */
export function LoginPage() {
  var _React$useState = React.useState(''),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      loginState = _React$useState2[0],
      setLoginState = _React$useState2[1];
  //loginState can be 'fetch-error', 'no-login', 'admin-login', and 'non-admin-login'.

  // /fetchAccessTokenValidity route requestwindow.
  // If access token is valid, admin cookie check and retrieval on client


  React.useEffect(function () {
    var ignore = false;
    setLoginState('');

    fetchAccessTokenValidity(setLoginState, ignore);

    return function () {
      ignore = true;
    };
  }, [setLoginState]);

  console.log('LoginPage() cookies on client:  ' + document.cookie + '.');

  return React.createElement(
    React.Fragment,
    null,
    React.createElement(
      'div',
      { className: 'top-bar' },
      React.createElement(
        'div',
        { className: 'top-bar-left-column' },
        React.createElement(
          'a',
          { href: window.frontendServerUrl, style: { textDecoration: 'none', color: 'rgb(255,255,255)' } },
          'Allen Young\'s Stockmarket Demo'
        )
      ),
      React.createElement(
        'div',
        { className: 'top-bar-right-column' },
        React.createElement(
          'a',
          { href: window.frontendServerUrl + '/registration.html', style: { textDecoration: 'none', color: 'rgb(255,255,255)' } },
          'Register'
        )
      )
    ),
    React.createElement(
      'section',
      { className: 'login-controls-section' },
      React.createElement(
        'h1',
        { className: 'login-controls-section-heading' },
        'Login'
      ),
      React.createElement(LoginControls, { loginState: loginState })
    ),
    footer()
  );
}

/* *************************************************************************************************************
LoginControls React component
************************************************************************************************************** */

export function LoginControls(_ref) {
  var loginState = _ref.loginState;

  var emailAddressInputRef = React.useRef(null);
  var passwordInputRef = React.useRef(null);
  var errorNoticeDomNodeRef = React.useRef(null);

  function handleShowHideClick(event) {
    event.preventDefault();
    event.stopPropagation();

    if (event.target.innerHTML === 'show') {
      event.target.innerHTML = 'hide';
      passwordInputRef.current.type = "text";
    } else {
      event.target.innerHTML = 'show';
      passwordInputRef.current.type = "password";
    }
  }

  function handleOnClick(event) {
    event.stopPropagation();

    errorNoticeDomNodeRef.current.innerHTML = '';

    emailAddressInputRef.current.style.backgroundColor = '#FFFFFF';
    passwordInputRef.current.style.backgroundColor = '#FFFFFF';

    /* ************************************************************************************************
    Input validation
    ************************************************************************************************ */
    var emailAddress = emailAddressInputRef.current.value.trim();
    if (!validateEmail(emailAddress)) {
      errorNoticeDomNodeRef.current.innerHTML = "Unable to log in.  You've entered an invalid email address.  Please correct and try again.";
      emailAddressInputRef.current.style.backgroundColor = '#FAA0A0';
      return;
    }

    var password = passwordInputRef.current.value.trim();
    if (password === '') {
      errorNoticeDomNodeRef.current.innerHTML = "Unable to log in.  You need to enter password.  Please correct and try again.";
      passwordInputRef.current.style.backgroundColor = '#FAA0A0';
      return;
    }

    if (!checkPasswordFormat(password)) {
      errorNoticeDomNodeRef.current.innerHTML = "Unable to log in.  Password does not meet the requirement.  Please correct and try again.";
      passwordInputRef.current.style.backgroundColor = '#FAA0A0';
      return;
    }

    /* ************************************************************************************************
    Login submission to backend
    ************************************************************************************************ */
    fetchLogin(emailAddress, password, errorNoticeDomNodeRef);
  }

  if (loginState === 'admin-login' || loginState === 'non-admin-login' || document.cookie.includes('accessToken=')) {
    return React.createElement(
      React.Fragment,
      null,
      React.createElement(
        'form',
        { className: 'login-form' },
        React.createElement(
          'div',
          { className: 'login-form-div-grid' },
          React.createElement(
            'label',
            { className: 'login-form-label-email-address', htmlFor: 'emailaddress' },
            'Email address'
          ),
          React.createElement('input', { disabled: true, className: 'login-form-input-email-address', type: 'text', id: 'emailaddress', name: 'emailaddress', size: '20', placeholder: 'Email address', ref: emailAddressInputRef }),
          React.createElement(
            'label',
            { className: 'login-form-label-password', htmlFor: 'password' },
            'Password'
          ),
          React.createElement('input', { disabled: true, className: 'login-form-input-password', type: 'password', id: 'password', name: 'password', size: '20', placeholder: 'Password', ref: passwordInputRef })
        ),
        React.createElement(
          'div',
          { className: 'login-form-div-login-button' },
          React.createElement('p', { className: 'login-form-error-notice', role: 'error-notice', ref: errorNoticeDomNodeRef }),
          React.createElement('input', { disabled: true, className: 'login-form-input-login-button', type: 'button', value: 'Login', onClick: handleOnClick })
        )
      )
    );
  } else {
    return React.createElement(
      React.Fragment,
      null,
      React.createElement(
        'form',
        { className: 'login-form' },
        React.createElement(
          'div',
          { className: 'login-form-div-grid' },
          React.createElement(
            'label',
            { className: 'login-form-label-email-address', htmlFor: 'emailaddress' },
            'Email address'
          ),
          React.createElement('input', { className: 'login-form-input-email-address', type: 'text', id: 'emailaddress', name: 'emailaddress', size: '20', placeholder: 'Email address', ref: emailAddressInputRef }),
          React.createElement(
            'label',
            { className: 'login-form-label-password', htmlFor: 'password' },
            'Password'
          ),
          React.createElement('input', { className: 'login-form-input-password', type: 'password', id: 'password', name: 'password', size: '20', placeholder: 'Password', ref: passwordInputRef }),
          React.createElement(
            'p',
            { className: 'login-form-password-show-hide' },
            React.createElement(
              'a',
              { href: '', role: 'password-show-hide', style: { textDecoration: 'none', color: 'rgb(255,255,255)' }, onClick: handleShowHideClick },
              'show'
            )
          )
        ),
        React.createElement(
          'div',
          { className: 'login-form-div-login-button' },
          React.createElement('p', { className: 'login-form-error-notice', role: 'error-notice', ref: errorNoticeDomNodeRef }),
          React.createElement('input', { className: 'login-form-input-login-button', type: 'button', value: 'Login', onClick: handleOnClick })
        )
      )
    );
  }
}