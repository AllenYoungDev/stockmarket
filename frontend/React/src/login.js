'use strict';

//import React, { StrictMode } from "react";
//import { createRoot } from 'react-dom/client';

import { validateEmail } from './utilities/validators.js';
import './settings/ServerUrls.js';
import {fetchAccessTokenValidity} from './utilities/fetchAccessTokenValidity.js';
import {fetchLogin} from './utilities/fetchLogin.js';
import { checkPasswordFormat } from './utilities/checkPasswordFormat.js';
import {footer} from './footer.js';


/* *************************************************************************************************************
LoginPage React component
************************************************************************************************************** */
export function LoginPage() {
  const [loginState, setLoginState] = React.useState('');
    //loginState can be 'fetch-error', 'no-login', 'admin-login', and 'non-admin-login'.

  // /fetchAccessTokenValidity route requestwindow.
  // If access token is valid, admin cookie check and retrieval on client
  React.useEffect(() => {
    let ignore = false;
    setLoginState('');

    fetchAccessTokenValidity(setLoginState, ignore);

    return () => {
      ignore = true;
    }
  }, [setLoginState]);

  console.log(`LoginPage() cookies on client:  ${document.cookie}.`)

  return (
    <React.Fragment>
    <div className="top-bar">
    <div className="top-bar-left-column"><a href={window.frontendServerUrl} style={{textDecoration: 'none', color: 'rgb(255,255,255)'}}>Allen Young's Stockmarket Demo</a></div>
    <div className="top-bar-right-column">
    <a href={window.frontendServerUrl + '/registration.html'} style={{textDecoration: 'none', color: 'rgb(255,255,255)'}}>Register</a>
    </div>
    </div>
    
    <section className="login-controls-section">
    <h1 className="login-controls-section-heading">Login</h1>
    <LoginControls loginState={loginState} />
    </section>

    {footer()}
    </React.Fragment>
  );
}

/* *************************************************************************************************************
LoginControls React component
************************************************************************************************************** */

export function LoginControls({loginState}) {
  const emailAddressInputRef = React.useRef(null);
  const passwordInputRef = React.useRef(null);
  const errorNoticeDomNodeRef = React.useRef(null);

  function handleShowHideClick(event) {
    event.preventDefault();
    event.stopPropagation();

    if (event.target.innerHTML === 'show') {
      event.target.innerHTML = 'hide';
      passwordInputRef.current.type = "text";
    } else {
      event.target.innerHTML = 'show';
      passwordInputRef.current.type = "password";
    }
  }

  function handleOnClick(event) {
    event.stopPropagation();

    errorNoticeDomNodeRef.current.innerHTML = ''  

    emailAddressInputRef.current.style.backgroundColor = '#FFFFFF';
    passwordInputRef.current.style.backgroundColor = '#FFFFFF';

    /* ************************************************************************************************
    Input validation
    ************************************************************************************************ */
    const emailAddress = emailAddressInputRef.current.value.trim();
    if(!validateEmail(emailAddress)) {
      errorNoticeDomNodeRef.current.innerHTML = "Unable to log in.  You've entered an invalid email address.  Please correct and try again."
      emailAddressInputRef.current.style.backgroundColor = '#FAA0A0';
      return
    }

    const password = passwordInputRef.current.value.trim();
    if(password === '') {
      errorNoticeDomNodeRef.current.innerHTML = "Unable to log in.  You need to enter password.  Please correct and try again.";
      passwordInputRef.current.style.backgroundColor = '#FAA0A0';
      return
    }

    if(!checkPasswordFormat(password)) {
      errorNoticeDomNodeRef.current.innerHTML = "Unable to log in.  Password does not meet the requirement.  Please correct and try again.";
      passwordInputRef.current.style.backgroundColor = '#FAA0A0';
      return
    }

    /* ************************************************************************************************
    Login submission to backend
    ************************************************************************************************ */
    fetchLogin(emailAddress, password, errorNoticeDomNodeRef);
  }
  
  if (loginState === 'admin-login' || loginState === 'non-admin-login' || document.cookie.includes('accessToken=')) {
    return (
      <React.Fragment>
      <form className="login-form">
        <div className="login-form-div-grid">
        <label className="login-form-label-email-address" htmlFor="emailaddress">Email address</label>
        <input disabled className="login-form-input-email-address" type="text" id="emailaddress" name="emailaddress" size="20" placeholder="Email address" ref={emailAddressInputRef} />  
        <label className="login-form-label-password" htmlFor="password">Password</label>
        <input disabled className="login-form-input-password" type="password" id="password" name="password" size="20" placeholder="Password" ref={passwordInputRef} />
        </div>

        <div className="login-form-div-login-button" >
        <p className="login-form-error-notice" role="error-notice" ref={errorNoticeDomNodeRef}></p>
        <input disabled className="login-form-input-login-button" type="button" value="Login" onClick={handleOnClick} />
        </div>
      </form>
      </React.Fragment>
    );
  } else {
    return (
      <React.Fragment>
      <form className="login-form">
        <div className="login-form-div-grid">
        <label className="login-form-label-email-address" htmlFor="emailaddress">Email address</label>
        <input className="login-form-input-email-address" type="text" id="emailaddress" name="emailaddress" size="20" placeholder="Email address" ref={emailAddressInputRef} />  
        <label className="login-form-label-password" htmlFor="password">Password</label>
        <input className="login-form-input-password" type="password" id="password" name="password" size="20" placeholder="Password" ref={passwordInputRef} />
        <p className="login-form-password-show-hide"><a href="" role="password-show-hide" style={{textDecoration: 'none', color: 'rgb(255,255,255)'}} onClick={handleShowHideClick}>show</a></p>
        </div>

        <div className="login-form-div-login-button" >
        <p className="login-form-error-notice" role="error-notice" ref={errorNoticeDomNodeRef}></p>
        <input className="login-form-input-login-button" type="button" value="Login" onClick={handleOnClick} />
        </div>
      </form>
      </React.Fragment>
    );    
  }
}