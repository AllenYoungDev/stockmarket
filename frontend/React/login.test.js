/**
 * @jest-environment jsdom
 */

//import React from "react";
//import ReactDOM from 'react-dom/client';

import renderer from 'react-test-renderer';
import {LoginPage, LoginControls} from './login.js';

jest.mock('./utilities/fetchAccessTokenValidity');

import {fetchAccessTokenValidity} from './utilities/fetchAccessTokenValidity';

(async () => {
  if (typeof React === 'undefined') {
    globalThis.React = await import("react");
  }
  if (typeof ReactDOM === 'undefined') {
    globalThis.ReactDOM = await import("react-dom/client");
  }
})();

var loginState = '';

async function loginPageJestSnapshotTest(loginStateArg) {
  let testRenderer;
  let testInstance;
  let tree;

  loginState = loginStateArg;

  //renderer.create() is wrapped in renderer.act() for executing useEffect() calls in <HomePage />.
  //https://github.com/facebook/react/issues/15321
  await renderer.act(async () => {
    testRenderer = renderer.create(
      <LoginPage />,
    );
  });

  testInstance = testRenderer.root;
  expect(testInstance.findByType(LoginControls).props.loginState).toBe(loginStateArg);

  tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
}

it('React app Jest snapshot testing for Allen Young\'s Stockmarket Demo login page', async () => {

  fetchAccessTokenValidity.mockImplementation((setLoginState, ignore) => {
    console.log('fetchAccessTokenValidity() mock called!');
    setLoginState(loginState);
  });

  /*************************************************************************************************
  No-login mock Jest snapshot test
  **************************************************************************************************/
  await loginPageJestSnapshotTest('no-login');

  /*************************************************************************************************
  Non-admin-user-login mock Jest snapshot test
  **************************************************************************************************/
  await loginPageJestSnapshotTest('non-admin-login');

  /*************************************************************************************************
  Admin-user-login mock Jest snapshot test
  **************************************************************************************************/
  await loginPageJestSnapshotTest('admin-login');

  /*************************************************************************************************
  Fetch-error mock Jest snapshot test
  **************************************************************************************************/
  await loginPageJestSnapshotTest('fetch-error');
});