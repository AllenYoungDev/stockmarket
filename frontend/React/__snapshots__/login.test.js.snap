// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo login page 1`] = `
[
  <div
    className="top-bar"
  >
    <div
      className="top-bar-left-column"
    >
      <a
        href="http://localhost"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Allen Young's Stockmarket Demo
      </a>
    </div>
    <div
      className="top-bar-right-column"
    >
      <a
        href="http://localhost/registration.html"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Register
      </a>
    </div>
  </div>,
  <section
    className="login-controls-section"
  >
    <h1
      className="login-controls-section-heading"
    >
      Login
    </h1>
    <form
      className="login-form"
    >
      <div
        className="login-form-div-grid"
      >
        <label
          className="login-form-label-email-address"
          htmlFor="emailaddress"
        >
          Email address
        </label>
        <input
          className="login-form-input-email-address"
          id="emailaddress"
          name="emailaddress"
          placeholder="Email address"
          size="20"
          type="text"
        />
        <label
          className="login-form-label-password"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="login-form-input-password"
          id="password"
          name="password"
          placeholder="Password"
          size="20"
          type="password"
        />
        <p
          className="login-form-password-show-hide"
        >
          <a
            href=""
            onClick={[Function]}
            role="password-show-hide"
            style={
              {
                "color": "rgb(255,255,255)",
                "textDecoration": "none",
              }
            }
          >
            show
          </a>
        </p>
      </div>
      <div
        className="login-form-div-login-button"
      >
        <p
          className="login-form-error-notice"
          role="error-notice"
        />
        <input
          className="login-form-input-login-button"
          onClick={[Function]}
          type="button"
          value="Login"
        />
      </div>
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      © 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo login page 2`] = `
[
  <div
    className="top-bar"
  >
    <div
      className="top-bar-left-column"
    >
      <a
        href="http://localhost"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Allen Young's Stockmarket Demo
      </a>
    </div>
    <div
      className="top-bar-right-column"
    >
      <a
        href="http://localhost/registration.html"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Register
      </a>
    </div>
  </div>,
  <section
    className="login-controls-section"
  >
    <h1
      className="login-controls-section-heading"
    >
      Login
    </h1>
    <form
      className="login-form"
    >
      <div
        className="login-form-div-grid"
      >
        <label
          className="login-form-label-email-address"
          htmlFor="emailaddress"
        >
          Email address
        </label>
        <input
          className="login-form-input-email-address"
          disabled={true}
          id="emailaddress"
          name="emailaddress"
          placeholder="Email address"
          size="20"
          type="text"
        />
        <label
          className="login-form-label-password"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="login-form-input-password"
          disabled={true}
          id="password"
          name="password"
          placeholder="Password"
          size="20"
          type="password"
        />
      </div>
      <div
        className="login-form-div-login-button"
      >
        <p
          className="login-form-error-notice"
          role="error-notice"
        />
        <input
          className="login-form-input-login-button"
          disabled={true}
          onClick={[Function]}
          type="button"
          value="Login"
        />
      </div>
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      © 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo login page 3`] = `
[
  <div
    className="top-bar"
  >
    <div
      className="top-bar-left-column"
    >
      <a
        href="http://localhost"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Allen Young's Stockmarket Demo
      </a>
    </div>
    <div
      className="top-bar-right-column"
    >
      <a
        href="http://localhost/registration.html"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Register
      </a>
    </div>
  </div>,
  <section
    className="login-controls-section"
  >
    <h1
      className="login-controls-section-heading"
    >
      Login
    </h1>
    <form
      className="login-form"
    >
      <div
        className="login-form-div-grid"
      >
        <label
          className="login-form-label-email-address"
          htmlFor="emailaddress"
        >
          Email address
        </label>
        <input
          className="login-form-input-email-address"
          disabled={true}
          id="emailaddress"
          name="emailaddress"
          placeholder="Email address"
          size="20"
          type="text"
        />
        <label
          className="login-form-label-password"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="login-form-input-password"
          disabled={true}
          id="password"
          name="password"
          placeholder="Password"
          size="20"
          type="password"
        />
      </div>
      <div
        className="login-form-div-login-button"
      >
        <p
          className="login-form-error-notice"
          role="error-notice"
        />
        <input
          className="login-form-input-login-button"
          disabled={true}
          onClick={[Function]}
          type="button"
          value="Login"
        />
      </div>
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      © 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo login page 4`] = `
[
  <div
    className="top-bar"
  >
    <div
      className="top-bar-left-column"
    >
      <a
        href="http://localhost"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Allen Young's Stockmarket Demo
      </a>
    </div>
    <div
      className="top-bar-right-column"
    >
      <a
        href="http://localhost/registration.html"
        style={
          {
            "color": "rgb(255,255,255)",
            "textDecoration": "none",
          }
        }
      >
        Register
      </a>
    </div>
  </div>,
  <section
    className="login-controls-section"
  >
    <h1
      className="login-controls-section-heading"
    >
      Login
    </h1>
    <form
      className="login-form"
    >
      <div
        className="login-form-div-grid"
      >
        <label
          className="login-form-label-email-address"
          htmlFor="emailaddress"
        >
          Email address
        </label>
        <input
          className="login-form-input-email-address"
          id="emailaddress"
          name="emailaddress"
          placeholder="Email address"
          size="20"
          type="text"
        />
        <label
          className="login-form-label-password"
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="login-form-input-password"
          id="password"
          name="password"
          placeholder="Password"
          size="20"
          type="password"
        />
        <p
          className="login-form-password-show-hide"
        >
          <a
            href=""
            onClick={[Function]}
            role="password-show-hide"
            style={
              {
                "color": "rgb(255,255,255)",
                "textDecoration": "none",
              }
            }
          >
            show
          </a>
        </p>
      </div>
      <div
        className="login-form-div-login-button"
      >
        <p
          className="login-form-error-notice"
          role="error-notice"
        />
        <input
          className="login-form-input-login-button"
          onClick={[Function]}
          type="button"
          value="Login"
        />
      </div>
    </form>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      © 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;
